add ssl patches to nginx-1.25 image for coroutines to work in lua client hello and cert ssl blocks

grounded042 opened the pull request:
<!--- Provide a general summary of your changes in the Title above --->
<!--- Please don't @-mention people in PR or commit messages (do so in an additional comment). --->
<!--- Please make sure you title is descriptive, it is used in the Release notes to let others know what it does ---> 

## What this PR does / why we need it:

In #11037 the base image was changed from `registry.k8s.io/ingress-nginx/nginx` to `registry.k8s.io/ingress-nginx/nginx-1.25`. This removed many patches to nginx which were previously included from the https://github.com/kubernetes/ingress-nginx/tree/main/images/nginx/rootfs/patches directory.

This PR adds openresty patches to the nginx-1.25 base image.

I originally opened this PR because of SSL patches, but in a Slack discussion
it was determined that including almost all patches openresty had for 1.25.3
was the best patch forward. Note that I did not include header and footer
patches which change headers and footers from `nginx` to `openresty`.

I renamed the patches to order them numerically as the don't apply correctly
if they are not applied in order.

I did not include the following patches:
- [builtin_error_page_footer.patch](https://github.com/openresty/openresty/blob/master/patches/nginx-1.25.3-builtin_error_page_footer.patch) since we don't want to patch in openresty to footers
- [server_header.patch](https://github.com/openresty/openresty/blob/master/patches/nginx-1.25.3-server_header.patch) since we don't want to patch in openresty to headers
- [setting_args_invalidates_uri.patch](https://github.com/openresty/openresty/blob/master/patches/nginx-1.25.3-setting_args_invalidates_uri.patch) as it's not needed [after nginx 1.7.1](https://github.com/openresty/openresty/blob/09efae3e6a51aa90228c58a179e7bbbdaf76ddf2/util/mirror-tarballs#L376-L381)
- [hash_overflow.patch](https://github.com/openresty/openresty/blob/master/patches/nginx-1.25.3-hash_overflow.patch) as it's not needed [after nginx 1.7.7](https://github.com/openresty/openresty/blob/09efae3e6a51aa90228c58a179e7bbbdaf76ddf2/util/mirror-tarballs#L397-L402)
- [no_pool.patch](https://github.com/openresty/openresty/blob/master/patches/nginx-1.25.3-no_pool.patch) as we don't want to [disable the nginx memory pool](https://github.com/openresty/no-pool-nginx)

<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here. -->

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] CVE Report (Scanner found CVE and adding report)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation only

## Which issue/s this PR fixes
<!--
(optional, in `fixes #<issue number>` format, will close that issue when PR gets merged):

fixes #
-->

## How Has This Been Tested?
I tested this locally via a simple nginx configuration. Before adding in the needed patches the `ssl_client_hello_by_lua_block` and `ssl_certificate_by_lua_block` would cause errors during `SSL_connect` (eg. `OpenSSL SSL_connect: SSL_ERROR_SYSCALL in connection to localhost:443`). After the patches were added no errors would occur.
```
error_log stderr info;

events {}
http {
	access_log stdout;

	# use docker embedded DNS
	resolver 127.0.0.11;

	server {
		listen 80;
		listen 443 ssl;

		ssl_certificate     /certs/rsa.crt;
		ssl_certificate_key /certs/rsa.key;

		ssl_client_hello_by_lua_block {
			ngx.log(ngx.INFO, "running: ssl_client_hello_by_lua_block")
			
			local http = require("resty.http")
			local httpc = http.new()
			httpc:set_timeout(1000)

			local res, err = httpc:request_uri("http://sidecar:8945", {
			  method = "GET",
			})
			if not res then
			  ngx.log(ngx.ERR, "client hello sidecar request failed: ", err)
			  return
			end
			
			ngx.log(ngx.INFO, "client hello sidecar request worked. got status code: ", res.status)
		}

		ssl_certificate_by_lua_block {
			ngx.log(ngx.INFO, "running: ssl_certificate_by_lua_block")
			
			local http = require("resty.http")
			local httpc = http.new()
			httpc:set_timeout(1000)

			local res, err = httpc:request_uri("http://sidecar:8945", {
			  method = "GET",
			})
			if not res then
			  ngx.log(ngx.ERR, "cert sidecar request failed: ", err)
			  return
			end
			
			ngx.log(ngx.INFO, "cert sidecar request worked. got status code: ", res.status)
		}

		location / {
			return 200 'hello, world';
		}
	}
}
```

<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I've read the [CONTRIBUTION](https://github.com/kubernetes/ingress-nginx/blob/main/CONTRIBUTING.md) guide
- [ ] I have added unit and/or e2e tests to cover my changes.
- [X] All new and existing tests passed.

Gacko commented:
/cc @strongjz 
/uncc @Gacko

tao12345666333 commented:
/assign

tao12345666333 reviewed (commented):
Thank you! You are right. When we upgraded the NGINX, we did not synchronize all patches.

grounded042 commented:
@tao12345666333 as we talked about in Slack I've updated this PR to include 1.25.3 patches from openresty. I've updated the description with these details as well, but some notes:

I renamed the patches to order them numerically as the don't apply correctly if they are not applied in order.

I did not include the following patches:
- [builtin_error_page_footer.patch](https://github.com/openresty/openresty/blob/master/patches/nginx-1.25.3-builtin_error_page_footer.patch) since we don't want to patch in openresty to footers
- [server_header.patch](https://github.com/openresty/openresty/blob/master/patches/nginx-1.25.3-server_header.patch) since we don't want to patch in openresty to headers
- [setting_args_invalidates_uri.patch](https://github.com/openresty/openresty/blob/master/patches/nginx-1.25.3-setting_args_invalidates_uri.patch) as it's not needed [after nginx 1.7.1](https://github.com/openresty/openresty/blob/09efae3e6a51aa90228c58a179e7bbbdaf76ddf2/util/mirror-tarballs#L376-L381)
- [hash_overflow.patch](https://github.com/openresty/openresty/blob/master/patches/nginx-1.25.3-hash_overflow.patch) as it's not needed [after nginx 1.7.7](https://github.com/openresty/openresty/blob/09efae3e6a51aa90228c58a179e7bbbdaf76ddf2/util/mirror-tarballs#L397-L402)
- [no_pool.patch](https://github.com/openresty/openresty/blob/master/patches/nginx-1.25.3-no_pool.patch) as we don't want to [disable the nginx memory pool](https://github.com/openresty/no-pool-nginx)

tao12345666333 commented:
Thank you! Let me take a look

tao12345666333 commented:
I'll merge this first, then initiate a new NGINX build. After that, we can verify if everything is working properly.

/lgtm
/approve

Gacko commented:
/cherry-pick release-1.10
/cherry-pick release-1.11

Gacko commented:
/cherry-pick release-1.11
